src/mpid/ch4/netmod/ofi/ofi_nic.h

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef OFI_NIC_H_INCLUDED
+#define OFI_NIC_H_INCLUDED
+
```

Comment:
This file need be `ofi_nic.c`. `ofi_nic.h` can contain just the prototypes this file exposes. I know it is more convenient to directly include the source, but it breaks the typical expectations thus adds more accidental complexities.

Reply:
Ok, that goes against the original goal. Originally, these were all in the same file as `ofi_init.c` as `static` functions to simplify the code and avoid symbol leak. Then they were `static inlines` in `ofi_nic.h`. Now they're going to have to be fully namespced in `ofi_nic.c/h`. I don't think we have an internal namespace for the OFI netmod, right?

Reply:
> I don't think we have an internal namespace for the OFI netmod, right?

No. Just use `MPIDI_OFI_`. It is less than ideal, but shouldn't be a blocker, right?

Reply:
Yeah, it's not that bad. Only some of the functions need to be in the header anyway. Just doing a quick compile check before I push the change.